### PR TITLE
Integrate update

### PR DESCRIPTION
**Pull request by 12ff54e**
* Iterate, instead of recursive, when adaptive integrating
* Add accuracy goal (though I name it prec in function parameter), as per #4

### CODE REVIEW

File: functions.h [hunk 1]
```diff
@@ -324,10 +315,10 @@ auto bessel_i_helper(const T& z) {
         y_tmp = 2. * n / z * y0 + y1;
         y1 = y0;
         y0 = y_tmp;
-        mu += 2. * (n % 2 == 0 ? 1 : -1) * (z.real() < 0 ? -1 : 1) * y1;
```
Comment (ssskkkky) [inline functions.h:327]: It seems to me this is totally different with before, but why we are still reasonable before?

File: functions.h [hunk 2]
```diff
@@ -324,10 +315,10 @@ auto bessel_i_helper(const T& z) {
...
+        mu += 2. * (std::real(z) < 0 ? 1 - 2 * (n & 1) : 1) * y1;
     }
 
-    mu = std::exp(z.real() < 0 ? -z : z) * (mu + y0);
```
Comment (ssskkkky) [inline functions.h:330]: like above

File: functions.h [hunk 3]
```diff
@@ -324,10 +315,10 @@ auto bessel_i_helper(const T& z) {
...
+    mu = std::exp(std::real(z) < 0 ? z : -z) * (mu + y0);
```
Comment (ssskkkky) [inline functions.h:321]: It seems that the negative sign can change only change the world just a little bit.

**Review by ssskkkky — APPROVED** [review]

File: functions.h (hunk 1 above)
Comment (12ff54e) [reply functions.h:327]: When z is small, this coefficient did not play an important role, I guess.

**Comment by 12ff54e** [comment]
This should also close #4

**Pull request merged**